src/main/java/com/diffblue/deeptestutils/mock/DTUMemberMatcher.java

```diff
@@ -0,0 +1,155 @@
+ * This implements TG-5254.
+ *
+ * @author <a href="http://diffblue.com">Diffblue</a>
+ */
+public final class DTUMemberMatcher {
+
+  /**
+   * Private constructor to prevent instantiation of the utility class.
+   */
+  private DTUMemberMatcher() {
+  }
+
+  /**
+   * Get a method matching the given name and parameter types. If there
+   * are multiple methods then pick the one that would be chosen by Java at
+   * runtime.
+   * <p>
+   * We first call the MemberMatcher of PowerMock to try and find a
+   * matching method. We return its result unless it throws a
+   * TooManyMethodsFoundException exception. In that case, we catch the
+   * exception and look for the method ourselves.
+   * <p>
+   * Java specification for method invocation:
+   * https://docs.oracle.com/javase/specs/jls/se8/html/jls-15.html#jls-15.12
+   *
+   * @param declaringClass The class to search
+   * @param methodName     The name of the desired method
+   * @param parameterTypes Types that can be fed as arguments to the method
+   *                       (can be more specific than the method's declared
+   *                       types)
+   * @return A matching method if such exists
+   * @throws MethodNotFoundException If a method cannot be found in the
+   *                                 hierarchy.
+   */
+  public static Method method(final Class<?> declaringClass,
+                              final String methodName,
+                              final Class<?>... parameterTypes) {
+
+    try {
+      return org.powermock.api.support.membermodification.MemberMatcher
+          .method(declaringClass, methodName, parameterTypes);
+    } catch (TooManyMethodsFoundException e) {
+      return methodUsingWholeSignature(declaringClass, methodName,
+          parameterTypes);
+    }
+  }
+
+  /**
+   * Get a method matching the given name and parameter types. If there
+   * are multiple methods then pick the one that would be chosen by Java at
+   * runtime.
+   * <p>
+   * We first gather all methods from the class and its ancestors that can
+   * be considered as a match and then find the best match using
+   * MethodComparatorFull.
+   * <p>
+   * Note that this does not handle generics, variadic arguments, arrays and
+   * possibly other cases correctly. For full spec on method invocation see
+   * https://docs.oracle.com/javase/specs/jls/se8/html/jls-15.html#jls-15.12
+   * <p>
+   * This is a helper method that is not intended to be called by users,
+   * instead use the "method" method.
+   *
+   * @param declaringClass The class to search
+   * @param methodName     The name of the desired method
+   * @param parameterTypes Types that can be fed as arguments to the method
+   *                       (can be more specific than the method's declared
+   *                       types)
+   * @return A matching method if such exists
+   */
+  protected static Method methodUsingWholeSignature(
```

Comment:
`protected` -> `private`

Reply:
@allredj if this becomes `private`, then we will not be able to unit test it.